src/utils.ts

```diff
@@ -6,3 +6,15 @@
 export async function sleep(ms: number): Promise<void> {
   return new Promise<void>(resolve => setTimeout(() => resolve(), ms))
 }
+
+export function getErrorMessage(error: unknown): string | undefined {
+  if (error instanceof Error) {
+    return error.message
+  }
+
+  if (typeof error === 'string') {
+    return error
+  }
+
+  return undefined
```

Comment:
Hmm instead of returning `undefined` I would suggest returning `String(error)`. You can throw other primitives/objects other than `string|Error` so we could at some point silent some problems, even though hopefully that won't happen as it is pretty disgusting 😅
See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw#description